gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -247,6 +250,94 @@ func (vm *VMKeeper) getGnoStore(ctx sdk.Context) gno.Store {
 	}
 }
 
+// Namespace can be either a user or crypto address.
+var reNamespace = regexp.MustCompile(`^gno.land/(?:r|p)/([a-zA-Z]+[_a-zA-Z0-9]+)`)
+
+// checkNamespacePermission check if the user as given has correct permssion to on the given pkg path
+func (vm *VMKeeper) checkNamespacePermission(ctx sdk.Context, creator crypto.Address, pkgPath string) error {
+	const sysUsersPkg = "gno.land/r/sys/users"
+
+	store := vm.getGnoStore(ctx)
+
+	// XXX: is this necessary ?
+	pkgPath = path.Clean(pkgPath) // cleanup pkgpath
```

Comment:
I would avoid changing the obtained path, It can be an attack vector from someone trying to spoof an user.